### ISSUE TITLE
Database cannot be saved on new device

### ISSUE THREAD

**Issue by Mithrandirro** [PhilippC/keepass2android#2459]
Version 1.09e-r7
on Samsung Galaxy S23Android 14
Before I used a Galaxy S10 on Android 12 (same K2A Version)
On the new phone when saving a database it says "Quelldatei konnte nicht gespeicert werden: Missing file..."
In English might be "Source file could not be saved: Missing file..."
Background:
The keepass database should be used onmy PC and on the smartphone. I use a synology NAS which synchronizes files between PC and NAS and I use FolderSynch to synch files between NAS and Smartphone.
It worked on the old Smartphone. Whenever I changed somthing I started the synch routines, so the same version of the file was on all devices and no matter where or what I changed, it worked.
With the new Smartphone I get the error message above."Quelldatei konnte nicht gespeicert werden: Missing file..."
I suppose K2A has no write access to the file, but I was not able to grant it anywhere.

**Comment by PhilippC** [comment]
instead of using FolderSync, I strongly recommend to access the NAS more "directly", e.g. through SFTP or WebDav (see e.g. https://kb.synology.com/de-de/DSM/tutorial/How_to_access_files_on_Synology_NAS_with_WebDAV#x_anchor_id5). This should avoid the problem you are seeing and has other advantages when it comes to synchronizing changes from different devices.

**Comment by Mithrandirro** [comment]
Can you elaborate a bit more why the issue exists?
The aim of my process using Folder synch is to copy the file from the NAS to the Smartphone **automatically** with a scheduled task. So I have the password file availabel offline from my NAS
On the old device I realized it with "synch me wireless" which is not compatible to the new smartphone.
My general question is, why does K2A care about the way I copy my files from the NAS to the device?
Or in other words, which property of the file, changed by "FolderSynch" is responsible for K2A not able to write updates?

**Comment by PhilippC** [comment]
> Can you elaborate a bit more why the issue exists?
Not really. I experience more and more issues with local file access on recent Android versions. Accessing files created by other apps is more and more restricted, and file access permissions might expire after updating the file by FolderSync.

> My general question is, why does K2A care about the way I copy my files from the NAS to the device?
I recommend not to copy them. Because FolderSync or any other third party tool cannot resolve conflicts (if you make a change on the phone while not connected to the internet and a change on the PC at the same time). Keepass2Android will store a local cache copy (so you *do* have offline access as well), but it can resolve the conflict by merging the files on an entry level.

**Comment by Mithrandirro** [comment]
> I recommend not to copy them. Because FolderSync or any other third party tool cannot resolve conflicts (if you make a change on the phone while not connected to the internet and a change on the PC at the same time). Keepass2Android will store a local cache copy (so you _do_ have offline access as well), but it can resolve the conflict by merging the files on an entry level.

I am aware of this issue and each time I change anything via Keepass (either on PC or Smartphone), I force a synch on the devices manually (on the PC it is even done automatically). And it worked properly so far.
Accessing the file online on the NAS is not an option. Might sound paranoid, but I do not want to be connected to the NAS constantly. I only connect to it via VPN.

I made some tests. If I export the database to a new file on the phone in the "documents" folder still the error occurs.
If I create a new one it works.

**Comment by PhilippC** [comment]
> Accessing the file online on the NAS is not an option. Might sound paranoid, but I do not want to be connected to the NAS constantly. I only connect to it via VPN.

If KP2A can go through this VPN (don't know how you set it up), this should work as well. But I understand you don't want to go that way.

> I made some tests. If I export the database to a new file on the phone in the "documents" folder still the error occurs.
If I create a new one it works.

I don't really understand what you are doing. Do you "export the database" from KP2A or from FolderSync?
